soroban-sdk-auth/src/lib.rs

```diff
@@ -8,13 +8,48 @@ use crate::public_types::{
     SignaturePayloadV0,
 };
 
+/// Users of this module will need to pass a struct that implements NonceAuth
+/// NonceAuth should manage a nonce in the contracts storage
```

Comment:
We need to expand this description a little. The NonceAuth also contains a signature, but that isn't discussed here.

Reply:
Updated